_scripts/config-fxios.js

```diff
@@ -1,53 +1,48 @@
 #!/usr/bin/env node
 
-var path = require("path");
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-var r = require("replace-in-file");
+const path = require('path');
+const fs = require('fs');
 
-var ios_path = process.env.FIREFOX_IOS_HOME;
-
-if (!ios_path) {
-  throw new Error("FIREFOX_IOS_HOME is not set");
+const iosPath = process.env.FIREFOX_IOS_HOME;
+if (!iosPath) {
+  throw new Error('FIREFOX_IOS_HOME is not set');
 }
 
-var config_path = path.join(
-  ios_path,
-  "Account",
-  "FirefoxAccountConfiguration.swift"
-);
+// This matches lines containing `FxAConfig(` with a `server: server` param.
+// We want to change this so `FxAConfig` is called with our `contentUrl`.
+const regex = /FxAConfig\([^)]*server: server[^)]*\)/g;
+const replaceFrom = /\bserver: server\b/;
+const replaceTo = 'contentUrl: "http://localhost:3030"';
+
+function replace() {
+  const filePath = path.join(iosPath, 'RustFxA', 'RustFirefoxAccounts.swift');
+  let fileContent = fs.readFileSync(filePath, 'utf8');
+  const matches = fileContent.match(regex);
 
-function replace(options) {
-  return new Promise(function(resolve, reject) {
-    r(options, function(err) {
-      if (err) reject(err);
-      resolve();
+  if (matches) {
+    matches.forEach((match) => {
```

Comment:
Why is a loop necessary?  I don't understand why we'd expect multiple matches here.

Reply:
Yeah I could remove, the whole thing just feels a little fragile so I tried to protect against changes somewhat. There is already [one conditional](https://github.com/mozilla-mobile/firefox-ios/blob/7e7aebb8166428b3b3635f59f63964ba7bc6aeee/RustFxA/RustFirefoxAccounts.swift#L137) around `config = FxAConfig` so this replaces them all if there are other matches in the future but as it stands now it's not necessary, true.